plugins/framework-qmk/fu-framework-qmk-device.c

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2025 Framework Computer Inc
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#include "config.h"
+
+#include "fu-framework-qmk-device.h"
+#include "fu-framework-qmk-struct.h"
+
+struct _FuFrameworkQmkDevice {
+	FuHidrawDevice parent_instance;
+	guint8 iface_reset;
+};
+
+G_DEFINE_TYPE(FuFrameworkQmkDevice, fu_framework_qmk_device, FU_TYPE_HIDRAW_DEVICE)
+
+#define FU_FRAMEWORK_QMK_RAW_USAGE_PAGE 0xFF60
+#define FU_FRAMEWORK_QMK_RAW_USAGE_ID	0x61
+
+static gboolean
+fu_framework_qmk_device_detach(FuDevice *device, FuProgress *progress, GError **error)
+{
+	FuFrameworkQmkDevice *self = FU_FRAMEWORK_QMK_DEVICE(device);
+	g_autoptr(GError) error_local = NULL;
+	g_autoptr(GByteArray) req = fu_struct_framework_qmk_reset_request_new();
+
+	if (!fu_hidraw_device_set_report(FU_HIDRAW_DEVICE(self),
+					 req->data,
+					 req->len,
+					 FU_IO_CHANNEL_FLAG_NONE,
+					 error)) {
+		g_prefix_error(error, "failed to write packet: ");
+		return FALSE;
+	}
+
+	/* success */
+	fu_device_add_flag(device, FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG);
+	return TRUE;
+}
+
+static void
+fu_framework_qmk_device_set_progress(FuDevice *self, FuProgress *progress)
+{
+	fu_progress_set_id(progress, G_STRLOC);
+	fu_progress_add_flag(progress, FU_PROGRESS_FLAG_GUESSED);
```

Comment:
can we removed guessed? it'll stop the big red message in the verbose logs.

Reply:
does that also mean i have to adjust the numbers? I just copied them from another plugin

Reply:
ok, done